devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -604,6 +652,80 @@ void YarpRobotLoggerDevice::unpackIMU(Eigen::Ref<const analog_sensor_t> signal,
     gyro = signal.segment<3>(6);
 }
 
+void YarpRobotLoggerDevice::lookForExogenousSignals()
+{
+    yarp::profiler::NetworkProfiler::ports_name_set yarpPorts;
+
+    auto time = BipedalLocomotion::clock().now();
+    auto oldTime = time;
+    auto wakeUpTime = time;
+    const auto lookForExogenousSignalPeriod = std::chrono::duration<double>(1);
+    m_lookForNewExogenousSignalIsRunning = true;
+
+    while (m_lookForNewExogenousSignalIsRunning)
+    {
+        // detect if a clock has been reset
+        oldTime = time;
+        time = BipedalLocomotion::clock().now();
+        // if the current time is lower than old time, the timer has been reset.
+        if ((time - oldTime).count() < 1e-12)
+        {
+            wakeUpTime = time;
+        }
+        wakeUpTime += lookForExogenousSignalPeriod;
+
+        bool allPortsConnected = true;
+        for (const auto& [name, signal] : m_vectorsCollectionSignals)
+        {
+            allPortsConnected = allPortsConnected && signal.connected;
+        }
+        for (const auto& [name, signal] : m_vectorSignals)
+        {
+            allPortsConnected = allPortsConnected && signal.connected;
+        }
+
+        if (!allPortsConnected)
+        {
+            // check for new messages
+            yarp::profiler::NetworkProfiler::getPortsList(yarpPorts);
```

Comment:
Why do you need the full list of ports if you already know the name of the port to connect to?

Reply:
to check if the port exists to establish the connection. maybe we can directly do the connection if not connected. I don't know if  we can improve the performances in this case

Reply:
https://github.com/ami-iit/bipedal-locomotion-framework/pull/570/commits/05342243cffc707ee03c605ce8a7baa1b3328d69 should implement what you suggested

Reply:
That's pretty neat!